src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -395,8 +385,8 @@ void ShenandoahControlThread::run_service() {
       last_shrink_time = current;
     }
 
-    // Don't wait around if there was an allocation failure - start the next cycle immediately.
-    if (!is_alloc_failure_gc()) {
+    // Don't wait if there was an allocation failure or another request was made mid-cycle.
```

Comment:
I'd rewrite the comment as "Wait for ShenandoahControlIntervalMax, unless there was an allocation failure or ..."